daffodil-cli/src/it/scala/org/apache/daffodil/schematron/TestEmbedded.scala

```diff
@@ -0,0 +1,105 @@
+
+import net.sf.expectit.matcher.Matchers.sequence
+import net.sf.expectit.matcher.Matchers.contains
+import org.junit.Test
+
+import java.util.UUID
+
+class TestEmbedded {
+  @Test def alwaysFails(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp(UUID.randomUUID.toString)
+    val schema = "xsd/always-fails-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> anyLines(2)
+  }
+
+  @Test def unitPriceWithoutValidation(): Unit = withShell() {
+    val data = mktmp("widget,monday,1,$5.00,$5.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list -s {{$schema}} $data" -> contains("</ex:list>\n")
+  }
+
+  @Test def unitPriceWithValidation(): Unit = withShell(FailureErrorCodeResult, true) {
+    val data = mktmp("widget,monday,1,$5.00,$6.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list --validate schematron={{$schema}} -s {{$schema}} $data" -> sequence(
+      contains("</ex:list>\n"),
+      anyLines(3)
+    )
+  }
+
+  @Test def unitPriceWithValidationCheckMessage(): Unit = withShell(FailureErrorCodeResult, true) {
+    val data = mktmp("widget,monday,5,$5.00,$25.00||gadget,tuesday,1,$10.00,$11.00")
+    val schema = "xsd/unit_price.dfdl.xsd"
+    s"parse -r list --validate schematron={{$schema}} -s {{$schema}} $data" -> sequence(
+      contains("</ex:list>\n"),
+      contains("[error] Validation Error: wrong unit price for gadget, tuesday\n"),
+      anyLines(2)
+    )
+  }
+
+  @Test def extends1(): Unit = withShell() {
+    val data = mktmp("bob;l;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends2(): Unit = withShell() {
+    val data = mktmp("ob;;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends3(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp(";;smith")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends4(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp("bob;l;")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def extends5(): Unit = withShell(FailureErrorCodeResult) {
+    val data = mktmp(";l;")
+    val schema = "xsd/extends-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</name>\n")
+  }
+
+  @Test def testWithNs1(): Unit = withShell() {
+    val data = mktmp("0;1")
+    val schema = "xsd/with-ns-1.dfdl.xsd"
+    s"parse --validate schematron={{$schema}} -s {{$schema}} $data" -> contains("</myns:interval>\n")
+  }
+
+  @Test def testWithNs2(): Unit = withShell(FailureErrorCodeResult) {
```

Comment:
Some of these tests check for FailureErrorCodeResults, but looked like they are checking for an infoset? Does that mean these tests fail validation? Is it worth also checking for a validation error in that case?

Reply:
that made for a nice refinement